### ISSUE TITLE
Prefetching and HTTPS

### ISSUE THREAD

**Issue by jeremyroman** [WICG/nav-speculation#124]
Chromium's implementation of speculation rules prefetch currently requires that URLs have the `https` scheme.

This requirement seems quite sensible in the case where IP anonymization is achieved with a proxy service (to avoid the traffic being visible to the egress node), but does this make sense for prefetch in general?

A few options:

**HTTPS only**: HTTPS is always required, even for non-anonymized prefetching. Authors should move to HTTPS going forward.

**Potentially trustworthy URLs only**: A [potentially trustworthy origin](https://www.w3.org/TR/secure-contexts/#is-origin-trustworthy) is required (or hypothetically a potentially trustworthy URL, if we want to enable pointless prefetching of `about:blank` and `data` URLs). Authors should move to HTTPS going forward, but localhost for development is fine too.

**No HTTPS restriction per se**: Prefetching from non-potentially trustworthy origins is acceptable if it is requested. User agents might, however, use this as an implementation-defined condition for dropping prefetches. For example, a UA that anonymizes IPs using a proxy service might only permit HTTPS traffic for public hosts, and nothing else.

My default preference is for no HTTPS restriction per se, I think. I could see an argument for potentially trustworthy origins only, because moving to that is a good idea regardless and while all plaintext HTTP traffic over the network is a potential security/privacy threat, this is an opportunity to remove some of it that doesn't break sites.

There's a related question of what should happen if a [localhost URL](https://datatracker.ietf.org/doc/html/draft-west-let-localhost-be-localhost) or a host that is on a [private network](https://wicg.github.io/private-network-access/#ip-address-space-heading) when resolved locally is requested with IP anonymity. I'd argue that certainly for localhost and possibly for private IPs, anonymity isn't possible if the anonymizing technology is a proxy service.

**Comment by domenic** [comment]
I agree that the choice is between "potentially trustworthy URLs only" and "no HTTPS restriction per se". Preventing testing of prefetching on localhost seems bad.

I lean toward potentially-trustworthy URLs only for the general avoidance-of-insecure-network-traffic reasons you mention. But I don't think there's any principled reason why prefetch is different than, say, img elements or iframes, so I don't think there's any clear argument that necessitates the restriction.

**Comment by jeremyroman** [comment]
Alright, let's see how potentially trustworthy goes. Easier to expand stuff later if necessary. If this looks good I'll see about filing a Chromium bug to allow potentially trustworthy non-HTTPS URLs (notably, localhost) when not proxied.
